src/api/app/helpers/webui/notification_helper.rb

```diff
@@ -9,17 +9,10 @@ def link_to_all
     end
   end
 
-  def filter_by_type_link(link_text, filter_item)
-    link_to(link_text, my_notifications_path(filter_item), class: filter_css(filter_item))
-  end
-
-  def filter_by_project_link(link_text, amount, filter_item)
-    badge_color = notification_filter_active?(filter_item) ? 'badge-light' : 'badge-primary'
+  def filter_notification_link(link_text, amount, filter_item)
```

Comment:
Codeclimate reported a high cognitive complexity here, so I splitted it up a bit